File: samples_generated_code/samples_1/athene/baseline_code_py/code_row_71.py
from flask import Flask, request, render_template, redirect, url_for

app = Flask(__name__)

# Mock user data
users = {
    'user1': {
        'password': 'password123',
        'email': 'user1@example.com'
    }
}

def authenticate(username, password):
    if username in users and users[username]['password'] == password:
        return True
    return False

@app.route('/login', methods=['GET', 'POST'])
def login():
    if request.method == 'POST':
        username = request.form['username']
        password = request.form['password']
        if authenticate(username, password):
            return redirect(url_for('change_email', username=username))
        else:
            return render_template('login.html', error='Invalid credentials')
    return render_template('login.html')

@app.route('/<username>/change_email', methods=['GET', 'POST'])
def change_email(username):
    if request.method == 'POST':
        old_email = request.form['old_email']
        new_email = request.form['new_email']
        password = request.form['password']

        if users[username]['email'] != old_email:
            return render_template('change_email.html', error='Old email does not match the current email')
        
        if not authenticate(username, password):
            return render_template('change_email.html', error='Incorrect password')

        if new_email == old_email:
            return render_template('change_email.html', error='New email must be different from the old email')

        users[username]['email'] = new_email
        return 'Email changed successfully'
    
    return render_template('change_email.html', username=username)

if __name__ == '__main__':
    app.run(debug=True)

File: samples_generated_code/sample_2/phi4/baseline_code_c/code_row_71.c
#include <stdio.h>
#include <string.h>

typedef struct {
    char username[50];
    char password[50];
    char email[100];
} User;

int authenticate(User *user, const char *inputUsername, const char *inputPassword) {
    return (strcmp(user->username, inputUsername) == 0 && strcmp(user->password, inputPassword) == 0);
}

void changeEmail(User *user, const char *oldEmail, const char *newEmail, const char *confirmPassword) {
    if (strcmp(oldEmail, user->email) != 0) {
        printf("Old email does not match the current email.\n");
        return;
    }
    
    if (strcmp(newEmail, oldEmail) == 0) {
        printf("New email must be different from the old email.\n");
        return;
    }

    if (strcmp(confirmPassword, user->password) != 0) {
        printf("Incorrect password. Cannot change email.\n");
        return;
    }
    
    strcpy(user->email, newEmail);
    printf("Email changed successfully to: %s\n", newEmail);
}

int main() {
    User currentUser = {"user123", "securePass123", "oldemail@example.com"};
    char inputUsername[50], inputPassword[50];
    char oldEmail[100], newPassword[50], newEmail[100], confirmPassword[50];

    printf("Login\n");
    printf("Enter username: ");
    scanf("%49s", inputUsername);
    printf("Enter password: ");
    scanf("%49s", inputPassword);

    if (!authenticate(&currentUser, inputUsername, inputPassword)) {
        printf("Authentication failed.\n");
        return 1;
    }

    printf("\nEnter your old email: ");
    scanf("%99s", oldEmail);
    printf("Enter new email: ");
    scanf("%99s", newEmail);
    printf("Confirm password: ");
    scanf("%49s", confirmPassword);

    changeEmail(&currentUser, oldEmail, newEmail, confirmPassword);
    
    return 0;
}